src/sdk/conference/channel.js

```diff
@@ -77,20 +77,20 @@ export class ConferencePeerConnectionChannel extends EventDispatcher {
 
   publish(stream, options) {
     if (options === undefined) {
-      options = {audio: !!stream.mediaStream.getAudioTracks(), video: !!stream
-          .mediaStream.getVideoTracks()};
+      options = {audio: !!stream.mediaStream.getAudioTracks().length, video: !!stream
+          .mediaStream.getVideoTracks().length};
     }
     if (typeof options !== 'object') {
       return Promise.reject(new TypeError('Options should be an object.'));
     }
     if (options.audio === undefined) {
-      options.audio = !!stream.mediaStream.getAudioTracks();
+      options.audio = !!stream.mediaStream.getAudioTracks().length;
     }
     if (options.video === undefined) {
-      options.video = !!stream.mediaStream.getVideoTracks();
+      options.video = !!stream.mediaStream.getVideoTracks().length;
     }
-    if (!!options.audio === !stream.mediaStream.getAudioTracks().length || !!
-    options.video === !stream.mediaStream.getVideoTracks().length) {
+    if ((!!options.audio == true && !!stream.mediaStream.getAudioTracks().length == false) ||
```

Comment:
`!!options.audio == true` could be `!!options.audio`, and `!!stream.mediaStream.getAudioTracks().length == false` could be `!stream.mediaStream.getAudioTracks().length`.

Reply:
thanks for your advice, I will change that.